### ISSUE TITLE
Non working distro-sync

### ISSUE THREAD

**Issue by praiskup** [rpm-software-management/dnf5#738]
I have `dnf5` installed from `dnf copr enable copr.fedorainfracloud.org/rpmsoftwaremanagement/dnf5-unstable`:

    dnf5-5.1.0-20230719020203.0.gc17c4f61.fc38.x86_64
    libdnf5-5.1.0-20230719020203.0.gc17c4f61.fc38.x86_64

I disabled the Copr repository, and I fail to "downgrade" the package using the
distro-sync command

    $ dnf copr disable copr.fedorainfracloud.org/rpmsoftwaremanagement/dnf5-unstable
    $ dnf copr distro-sync dnf5

I can not even downgrade for that matter...

    $ dnf downgrade dnf5
    Problem: conflicting requests
      - cannot install both dnf5-5.0.13-2.fc38.x86_64 and dnf5-5.1.0-20230719020203.0.gc17c4f61.fc38.x86_64
      - cannot install both dnf5-5.0.6-2.fc38.x86_64 and dnf5-5.1.0-20230719020203.0.gc17c4f61.fc38.x86_64
      - problem with installed package 

(I have alias dnf=dnf5)

**Comment by evan-goode** [comment]
Likely another issue caused by us changing the default value of `allow_vendor_change` to `false` (https://github.com/rpm-software-management/dnf5/issues/722, https://github.com/rpm-software-management/dnf5/issues/701, https://github.com/rpm-software-management/dnf5/pull/623).

Does it work with `--setopt=allow_vendor_change=true`?

**Comment by j-mracek** [comment]
The question is whether we should revert `allow_vendor_change`  to `true`.
Basically there are several problems
- command line packages are refused to get installed when they have a different vendor
  - This could be fixed
- disabling copr repository results in refusing installation of packages from original repositories - skipping all upgrades
  - I thing it is not fixable, because it is the principle of the feature
- cryptic solver message
  - it is impossible to guess that the real issue is related to a different vendor of both packages
  - `- cannot install both dnf5-5.0.13-2.fc38.x86_64 and dnf5-5.1.0-20230719020203.0.gc17c4f61.fc38.x86_64`

@Conan-Kudo @dcantrell @kontura @evan-goode @m-blaha @jrohel @pkratoch @inknos  I think it would be better to revert the default of `allow_vendor_change` for now to decrease pressure on DNF5 Fedora 39 release. What do you think?

**Comment by m-blaha** [comment]
I totally agree with going back to `true`. The most concerning issue (and what currently renders setting `allow_vendor_change=false` unusable) is that cryptic error message part. Even we as devs (or at least me) are just guessing what's the root cause, based on similar errors reported recently. I guess fixing this will require some adjustments in libsolv's problem reporting, right?

**Comment by praiskup** [comment]
> Does it work with --setopt=allow_vendor_change=true?

Ah, yes, I forgot.

> cryptic error message part

Agreed, but here actually was no error message, just "Nothing to do".  So +1 for being more explicit. E.g:

    Not {upgrading,downgrading,...} because of RPM Vendor change:
        dnf5    "Fedora Copr - user praiskup" => "Fedora Project"
        libdnf5 "Fedora Copr - user praiskup" => "Fedora Project"
        ...
    Nothing to do.

**Comment by m-blaha** [comment]
> Agreed, but here actually was no error message, just "Nothing to do". So +1 for being more explicit. E.g:
> 
> ```
> Not {upgrading,downgrading,...} because of RPM Vendor change:
>     dnf5    "Fedora Copr - user praiskup" => "Fedora Project"
>     libdnf5 "Fedora Copr - user praiskup" => "Fedora Project"
>     ...
> Nothing to do.
> ```

Oh yeah, this even worse than `cannot install both...` message.

**Comment by Conan-Kudo** [comment]
The information should exist from libsolv, since Zypper is able to report it as @m-blaha shows. So I guess, what are we missing to be able to expose that?
